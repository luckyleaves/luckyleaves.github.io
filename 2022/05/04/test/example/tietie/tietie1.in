1000
886507892 378105888 540661421 198385036 89815037 119528681 197193806 710522180 663644611 422344428 706104238 939009182 263643943 402126097 454913277 564368052 141652019 733595172 530721753 506948951 248424637 280155585 378105888 432379663 661436082 9691791 167689123 234043876 348007645 259619775 153987203 391262605 409744900 323836912 360873012 388680563 340006050 604950214 795270586 25540109 655803524 917429395 4981518 122912169 686630279 950243199 243540087 825868494 983733800 705864842 922883890 338565101 570810971 111439947 530721753 646862680 870353899 786963594 413681957 287118025 65886723 606460664 910647919 372850668 983733800 801117782 338596875 153987203 706578908 922888167 868010084 825868494 381432016 494414378 409744900 804076752 138102146 414414844 943032432 221287162 505550075 648807753 282996278 311153771 452623980 942959741 153377131 311153771 122295284 34617716 538934893 971653089 981262952 514693799 76038446 853563816 433717347 144366159 19438940 562492325 851837774 296807241 903868393 470981951 200721972 598344057 209583285 703244378 945117942 425660241 280155585 707904063 619330733 903095778 273920029 968194450 495599936 351709054 511094737 883201230 424645970 776847628 886507892 808597032 542899345 287118025 770121283 2253630 514693799 470141766 288818792 173104114 719479245 464983681 89815037 939009182 389538815 624037926 712902938 233221168 150039934 210937309 677676092 331621767 850876171 837559599 336092629 853563816 40014755 804536816 710522180 901735722 720105433 795080264 495599936 354960068 720105433 954681685 340006050 236983628 119528681 795270586 916590577 323191186 269213889 369910864 565504973 649749085 983733800 121466928 847779502 829282233 583209791 513400716 791005612 125126243 686003097 83525001 786963594 504803984 111439947 759341719 246481933 903095778 323836912 243540087 330860297 94629168 323836912 675877779 974916938 986964645 618699765 943032432 439369497 416962214 82702433 272569362 837233284 677676092 723266866 47110562 850876171 247814194 424645970 103336342 49642394 648807753 161467739 189091827 927249446 597101363 455248786 727057277 913007407 23483227 830575970 34617716 311153771 926469612 597537584 751186962 462306613 846279679 227959091 877295931 19438940 619330733 862691862 351709054 606669618 81461555 95404073 591804710 712902938 248424637 384333818 220908590 973532100 151513705 795080264 886507892 778031346 170807275 939009182 400419282 348007645 338565101 372850668 391089769 433717347 630928743 454879273 85358207 309328216 336092629 330279520 350180379 993302525 850876171 608929071 973532100 703716473 426955700 885672108 131861134 95404073 427679118 597101363 273920029 391089769 569291937 717736996 210572480 121466928 929490956 334400118 656010534 49642394 676512044 696702376 205164139 138102146 618699765 810545295 680696486 853458516 10351673 649220147 465867985 505550075 606460664 391262605 574707408 542139415 40014755 425660241 117347049 119528681 230124780 813133067 763433150 331214270 726821663 695990678 565504973 219749387 656010534 179792254 586136291 3627453 644328998 70508376 370777545 903095778 293448279 910076026 910891615 903095778 247814194 661436082 461873918 92011799 221287162 196806699 890513112 260337660 653393259 125126243 834515006 801117782 754409194 462880594 795270586 981262952 251367975 221287162 681431438 597101363 94629168 850876171 42437941 804076752 263643943 263937889 642308582 125587844 454913277 570810971 870353899 96230782 699884524 680696486 263643943 323191186 424645970 200721972 795080264 464152304 42175553 263937889 727057277 247299743 462880594 950243199 971653089 966416996 476893734 409744900 853458516 983495605 210286849 494414378 197193806 837233284 918268867 829282233 570810971 875143409 496423643 247814194 330860297 476893734 315133179 266753953 749270921 452623980 968194450 586136291 153377131 258978194 121758800 974916938 502299091 233221168 504803984 40387084 629959642 776847628 885672108 486095973 722683074 400975830 452623980 227959091 744621948 696302967 330860297 447980724 365255915 227959091 651940945 288818792 602878963 125587844 101698743 922883890 70508376 542139415 161467739 471607532 831786707 273682754 706578908 498311372 210286849 229198662 323191186 167689123 23483227 604988188 897478149 853673230 897478149 445087070 371365371 363236785 744621948 694149735 461873918 3627453 110653872 263643943 80900903 689172882 409744900 916590577 388680563 358315875 954831507 371365371 639686479 950452902 414732556 897478149 596628580 586136291 791005612 25498677 440636655 400419282 502299091 759341719 462306613 84860509 331214270 413681957 96230782 504803984 651936026 391262605 272569362 80900903 338565101 928153183 340006050 875659898 450792064 101698743 34617716 705864037 513400716 424645970 334400118 236983628 391262605 55351017 574279962 254578783 569291937 476893734 34617716 754501637 913007407 517403523 207391244 247814194 516767231 269213889 903095778 922883890 330279520 443096173 813133067 229198662 205684245 89815037 595357247 983733800 804076752 966416996 950452902 686630279 967902794 875143409 323397673 862691862 875143409 571463337 496423643 626967958 759341719 424645970 703244378 76642006 621624771 502171374 633691177 896094359 760227353 323191186 65886723 378105888 251367975 289624046 557049439 372850668 462880594 323191186 221287162 336092629 9691791 164911066 516767231 229198662 945117942 850876171 721098044 13178873 630928743 754409194 92703824 515174829 410689934 804536816 271173938 530073805 653393259 296645847 847241645 229101235 323836912 373465667 223816513 984831832 791005612 630928743 887526063 712814440 254578783 816412195 414732556 632780766 602878963 457391153 922883890 706578908 655591024 760876728 862691862 661436082 271173938 811075011 538174071 719479245 467681207 929490956 642308582 829012793 862691862 762196572 141652019 928153183 813361600 25540109 164911066 269213889 372850668 498311372 850876171 277524073 81461555 246063366 144366159 50277537 687139684 311153771 399377686 707904063 875143409 774943304 618699765 604950214 706104238 847241645 626867182 712902938 817175132 336317140 259246019 652720066 520739074 170807275 351709054 502299091 55351017 96230782 571463337 621624771 846279679 424645970 440636655 574279962 22452125 606460664 877295931 360531211 868010084 210572480 921961917 939009182 595357247 946075497 899902511 475166413 817175132 5485802 773803429 726821663 615783247 770121283 591804710 943032432 651940945 365255915 837233284 562492325 847779502 358315875 433717347 49895856 338596875 722746594 922888167 399377686 330279520 309328216 19438940 419508367 699171161 754501637 693006694 470141766 862691862 358315875 290345282 495599936 158727675 372850668 223816513 623864291 604988188 926656404 363236785 642652036 462306613 219749387 260337660 229198662 158727675 516879982 881627213 738185877 205684245 138102146 529840948 515174829 945117942 571463337 40387084 263643943 574707408 31381991 364075262 630270616 973138125 404325142 230124780 82744364 13147234 47110562 699884524 216790596 811075011 574707408 791005612 350633464 981262952 795270586 303600308 540661421 145412712 564368052 530073805 986964645 602878963 404325142 369910864 586897240 974916938 597101363 124683179 42437941 264628195 619330733 687139684 983733800 628777239 608929071 125126243 364075262 881627213 388680563 642652036 151513705 49895856 646862680 505550075 502171374 56383905 564368052 271173938 885672108 342693854 827341843 125126243 883856075 93192405 583209791 817175132 763203856 439369497 810545295 515174829 689172882 416962214 911653678 853563816 516879982 886507892 464983681 311153771 207546419 439369497 91595891 830912295 666196432 531466067 713332576 432379663 763269287 555285252 703113882 452623980 717736996 745560659 927249446 853458516 273682754 461873918 21754144 555428052 330279520 277524073 432379663 259846697 762196572 137689166 877295931 53027678 631175501 151513705 230124780 964453087 771275492 364075262 467681207 261926459 331214270 137689166 887526063 804536816 897478149 683293634 525163619 945117942 911653678 23483227 141652019 470981951 628777239 494414378 582002680 582002680 596628580 756586556 916967043 583209791 331974091 378105888 986964645 360531211 586136291 883856075 369910864 978275142 25498677 675185852 391262605 236632750 209583285 929419901 428187439 916967043 422344428 733595172 870353899 585485794 95404073 833143579 528040568 875143409 899902511 336092629 649220147 35401716 829282233 84860509 918268867 241730714 138102146 210572480 763269287 360873012 885483703 679808606 315133179 628777239 595357247 462306613 230124780 365255915 732289187 354960068 829282233 812818233 225800187 22452125 338660823 927249446 615783247 296056074 424645970 183895162 419508367 681431438 598344057 297720717 890513112 130932621 305026862 220876583 151513705 903095778 372850668 699171161 569482518 897478149 723266866 269213889 656010534 70508376 312032732 899902511 96230782 910647919 465867985 883856075 305026862 42175553 649749085 513400716 808597032 287118025 597101363 288818792 219749387 455711671 726821663 706578908 96832833 604988188 720105433 40387084 242551049 40387084 754409194 137129254 219749387 53027678 93192405 470981951 1594151 513375640 196806699 138102146 695990678 254578783 560444599 937862048 651940945 388166534 29180212 973138125 686630279 364075262 234043876 272569362 875143409 911653678 754501637 674269808 32206321 666196432 323191186 875143409 899902511 531466067 76642006 297125890 829282233 984831832 926469612 432379663 640065878 922888167 922883890 320332327 817175132 720105433 618699765 92703824 687139684 400975830 732565170 360873012 528040568 258978194 198385036 733006257 917429395 351709054 272569362 122912169 49895856 681431438 513400716 811075011 575596695 837233284 196806699 59717689 164911066 249709224 804536816 229101235 710522180 290345282 771275492 3627453 
1000
101 989
277 304
202 379
125 711
458 908
272 598
242 262
504 912
431 436
353 610
193 452
94 447
122 402
178 996
280 984
479 891
7 148
9 496
596 680
576 628
224 453
262 469
39 275
505 784
171 605
190 247
99 195
576 706
56 75
386 852
286 588
487 872
324 814
183 474
327 548
41 168
639 744
410 597
285 680
160 270
28 321
242 281
679 822
179 583
297 950
510 588
459 597
332 945
307 633
467 978
519 985
258 504
643 700
525 878
543 575
234 637
111 310
81 904
250 730
15 973
592 750
755 878
420 428
97 155
188 937
492 563
177 905
317 702
31 543
397 722
479 982
50 579
373 625
740 939
124 838
651 719
248 653
113 205
456 899
311 542
270 500
760 943
186 591
350 972
423 497
467 617
288 825
310 832
206 379
607 750
93 590
198 485
462 670
326 659
102 624
115 989
583 894
432 916
336 446
443 644
263 285
188 417
288 289
96 166
322 962
20 460
112 146
92 765
624 730
377 952
40 675
363 614
700 906
443 906
528 887
65 330
315 723
14 537
621 638
429 891
711 743
262 270
477 899
477 834
648 881
419 869
582 622
346 931
153 924
214 412
561 564
15 757
474 794
330 509
130 255
35 677
450 698
382 412
365 463
285 627
234 597
576 966
174 332
78 973
342 853
150 992
246 823
377 948
347 380
39 455
474 857
787 911
910 990
917 977
221 473
502 647
24 837
213 822
366 871
81 854
334 452
743 928
21 317
297 364
749 910
53 604
511 654
59 696
28 994
252 860
185 704
404 406
692 794
352 991
410 481
292 905
71 315
352 823
294 590
447 466
347 423
400 482
249 641
364 572
230 747
41 952
195 739
270 611
266 737
100 346
339 912
51 820
132 671
272 683
27 452
574 579
210 589
658 763
531 788
280 668
401 687
297 665
249 473
222 379
271 866
416 945
460 836
75 125
231 526
446 932
409 962
195 611
596 829
106 768
326 868
285 286
675 687
465 600
20 158
39 245
514 784
610 669
461 714
195 679
333 515
70 456
325 331
188 672
361 418
100 365
118 887
550 807
353 617
439 675
335 467
26 111
350 704
71 350
155 968
61 844
673 860
721 822
551 997
206 605
375 724
226 842
189 930
73 188
515 837
224 1000
171 815
313 415
343 992
93 361
190 901
822 939
218 769
367 942
243 337
701 769
688 862
387 830
407 824
142 809
291 485
96 630
730 899
263 607
217 417
653 686
439 757
347 450
480 862
142 988
73 578
655 902
868 967
106 190
739 857
226 849
49 687
527 561
527 978
223 554
223 781
267 796
113 202
829 906
465 864
367 538
582 858
792 874
313 419
598 622
68 456
373 395
135 569
746 835
149 901
842 850
676 851
399 936
644 671
125 625
233 787
335 468
88 253
305 649
405 938
829 887
148 481
531 690
312 712
260 557
261 490
641 904
181 874
307 405
14 849
128 356
621 983
427 938
749 852
832 976
16 640
291 508
382 843
40 811
66 246
236 680
271 295
702 715
80 557
25 745
249 324
109 487
869 932
156 364
46 343
144 874
624 948
160 387
654 836
46 468
35 39
469 763
569 572
327 985
222 872
612 929
653 921
179 832
618 962
234 964
354 489
326 423
174 455
403 646
78 977
313 694
143 197
132 691
252 364
109 689
181 583
444 836
157 715
223 476
478 659
20 756
827 953
73 963
513 579
331 685
508 525
413 571
580 719
418 480
109 204
37 201
334 651
607 905
424 612
667 813
404 631
266 340
241 400
40 48
12 757
762 962
765 846
347 702
65 115
961 983
35 886
541 659
72 238
285 350
208 539
776 801
144 795
125 130
928 932
391 951
693 738
12 696
185 955
447 953
226 532
447 860
675 835
43 194
333 939
404 592
801 983
11 872
391 859
45 512
310 888
121 169
513 522
159 280
257 751
384 969
164 576
169 878
168 930
632 778
469 707
384 421
62 747
610 617
860 980
649 734
852 918
873 911
41 525
879 900
6 816
148 337
449 633
263 924
702 735
147 616
290 591
295 402
672 913
423 628
162 917
99 547
90 791
649 882
490 995
491 632
504 646
703 714
339 684
207 678
453 686
347 391
437 580
236 328
148 700
530 876
361 607
316 458
321 496
90 722
105 741
545 735
255 668
429 877
298 399
103 893
731 942
506 687
631 645
171 496
291 380
818 998
374 514
258 731
29 550
358 985
601 807
512 919
457 470
663 961
415 836
140 385
136 605
37 546
855 870
587 618
242 550
285 639
533 824
396 852
513 676
13 818
54 642
688 978
797 848
213 746
850 930
24 208
17 813
116 325
560 957
442 658
435 442
552 826
130 268
243 355
569 633
254 905
526 655
44 668
683 814
921 984
389 973
351 501
129 485
661 962
139 765
791 855
876 972
334 857
489 512
624 880
302 389
182 777
626 892
70 133
293 979
396 488
158 498
130 177
155 712
652 823
769 796
719 992
626 983
339 912
139 705
732 740
415 634
89 760
255 927
235 621
446 502
701 729
685 856
127 297
60 981
103 836
831 983
403 857
344 357
146 913
107 532
89 746
250 502
325 860
93 858
107 385
358 661
242 613
119 899
203 292
299 648
185 565
626 960
313 814
146 778
494 897
427 916
65 171
61 997
491 520
284 631
82 224
301 786
313 339
804 867
161 443
112 207
612 912
406 604
129 671
11 219
244 323
232 273
420 735
777 971
163 797
126 450
428 628
68 810
198 513
29 831
169 515
59 193
111 494
84 950
8 198
920 967
105 627
6 573
447 858
12 311
713 723
400 821
231 909
212 497
321 363
284 958
101 798
497 800
236 671
378 873
445 780
651 837
748 903
745 954
628 817
67 608
585 960
810 958
11 230
75 540
3 662
52 345
425 852
645 817
43 151
68 878
142 996
187 334
564 1000
592 741
892 981
536 655
199 835
770 868
236 253
193 356
754 769
21 103
164 965
333 714
546 908
382 849
108 710
518 959
95 262
610 855
465 925
179 643
492 494
249 799
362 699
341 792
726 961
99 976
211 803
32 407
282 884
74 398
756 974
199 629
86 416
55 273
171 913
60 491
267 457
649 861
470 887
582 961
267 448
503 948
31 291
842 960
120 597
428 945
139 346
670 899
122 136
68 378
892 968
43 493
134 162
781 954
357 402
60 236
528 860
129 591
222 726
721 988
208 208
349 855
82 933
351 394
327 943
147 699
54 598
196 607
141 482
27 53
503 543
915 922
265 490
552 953
10 93
20 65
722 787
142 423
231 949
171 518
749 976
339 904
471 508
705 819
345 454
424 427
603 980
40 177
248 493
459 721
13 284
263 407
143 734
548 725
573 874
25 815
200 297
917 939
134 289
468 955
453 873
418 604
80 302
349 603
147 247
574 816
85 213
23 59
88 834
37 95
503 625
551 577
158 231
318 892
58 884
314 321
324 841
227 644
610 672
43 735
292 883
511 629
68 484
349 837
220 699
445 797
552 695
495 581
290 867
251 671
50 551
201 753
666 887
168 912
46 77
158 660
23 995
178 569
106 160
486 742
270 564
452 565
307 974
446 640
186 601
173 519
172 213
290 932
391 697
333 849
406 652
34 202
422 685
39 267
516 768
33 261
855 868
535 909
160 335
63 642
611 724
389 909
61 407
373 611
225 596
763 770
207 373
522 880
265 462
1 863
327 505
497 510
736 856
866 875
407 792
526 838
22 581
499 803
8 895
143 283
448 878
470 552
702 807
282 948
245 329
38 239
89 683
339 914
348 883
305 845
177 840
390 888
205 366
55 802
488 569
198 222
214 231
147 408
522 902
165 956
113 140
737 777
531 536
194 682
741 928
737 770
80 96
5 220
487 731
231 611
347 391
43 226
112 674
68 622
217 634
65 926
11 119
57 596
275 764
662 929
83 369
511 723
659 756
442 497
249 397
250 464
717 832
66 746
862 910
515 896
723 818
337 854
99 189
506 691
223 557
298 561
261 906
27 461
357 939
152 623
351 834
211 451
166 623
77 105
263 896
66 451
118 922
82 921
252 773
506 798
124 926
552 745
716 992
765 871
723 909
187 632
154 327
143 567
650 876
700 807
149 418
888 915
503 556
662 779
259 310
659 822
37 200
146 390
350 468
224 709
665 847
691 694
333 504
509 803
454 954
87 351
302 920
445 741
247 450
510 993
117 409
662 850
352 620
309 688
550 953
335 624
405 479
306 763
85 103
728 957
57 636
2 63
250 453
145 931
221 608
125 591
255 493
456 778
699 977
883 891
565 725
36 960
190 711
239 447
399 853
234 487
62 290
271 342
25 417
305 398
930 931
303 487
121 392
354 590
76 924
235 681
253 466
504 515
376 981
24 213
618 814
132 890
404 477
34 486
221 251
16 706
162 488
6 248
770 812
35 725
447 473
282 515
359 629
709 795
92 194
571 953
11 794
58 453
444 596
56 764
236 708
570 862
472 845
52 110
106 713
247 568
110 508
422 886
212 232
55 763
474 538
24 568
702 932
610 852
218 809
285 647
639 790
745 979
425 605
733 834
128 557
