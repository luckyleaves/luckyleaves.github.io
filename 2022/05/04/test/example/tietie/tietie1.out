1
As we can
4
2
1
2
As we can
1
2
2
2
2
4
1
1
1
4
2
16
10
2
2
4
3
2
As we can
4
10
As we can
1
2
1
2
2
2
4
12
2
2
4
4
21
12
2
1
3
3
1
2
1
1
2
12
1
17
2
4
1
2
1
12
1
As we can
38
1
3
1
2
2
2
1
2
2
1
1
12
2
4
1
2
2
1
2
1
2
3
1
1
4
12
2
2
3
2
2
1
1
1
2
3
As we can
4
As we can
4
1
2
As we can
2
12
1
2
2
1
1
1
4
2
2
As we can
1
As we can
As we can
1
1
1
1
As we can
1
1
4
As we can
2
3
2
4
2
3
6
2
2
2
1
4
1
1
1
2
1
29
2
1
1
2
2
2
3
1
2
1
1
2
1
4
4
1
2
3
2
1
1
2
As we can
12
1
2
1
4
2
2
As we can
6
2
2
2
2
1
2
2
2
4
1
2
2
2
2
As we can
2
12
8
2
2
2
2
4
1
1
1
4
2
1
1
2
1
2
1
As we can
As we can
3
4
4
3
28
3
2
2
2
As we can
2
6
4
1
10
2
3
2
4
2
4
1
1
1
12
1
2
2
1
1
4
1
1
2
4
1
4
1
1
2
1
4
13
1
1
1
2
2
2
1
2
4
16
3
2
1
1
2
1
2
4
1
1
2
8
1
2
2
2
4
18
1
2
1
1
4
As we can
2
As we can
2
1
1
As we can
1
1
As we can
2
2
2
4
2
1
18
2
8
2
2
2
1
1
4
1
4
1
1
1
2
2
2
1
2
4
2
As we can
13
2
2
4
2
2
4
4
1
1
4
1
2
As we can
3
As we can
1
1
1
1
1
1
1
2
6
2
2
1
2
4
2
4
2
2
1
2
2
3
2
2
1
3
2
3
2
10
2
4
4
2
1
2
12
2
4
4
As we can
2
1
1
2
4
As we can
1
10
4
4
2
15
2
As we can
As we can
1
13
2
1
1
2
1
1
4
1
2
1
1
1
2
1
4
3
4
2
1
2
1
1
12
3
6
2
As we can
2
12
18
18
2
As we can
2
4
3
1
13
2
2
4
1
2
1
2
2
1
1
3
3
As we can
2
2
3
15
3
4
2
1
2
2
2
2
2
10
2
1
4
1
1
3
As we can
2
4
1
2
2
2
1
12
1
As we can
1
1
4
2
2
As we can
As we can
2
2
1
1
3
2
2
1
1
2
2
4
2
4
1
3
As we can
1
4
4
10
1
8
2
12
2
1
2
2
1
2
1
2
1
As we can
1
4
2
1
4
1
2
2
4
2
12
15
1
1
1
2
As we can
2
2
1
2
12
13
1
4
1
1
2
1
10
1
2
2
2
1
1
4
2
2
1
21
2
2
1
2
2
1
1
4
1
3
2
4
2
4
1
2
4
1
2
2
4
4
21
2
1
2
2
2
2
2
1
2
4
2
1
4
2
2
2
1
4
As we can
2
1
2
10
1
2
3
2
1
3
1
1
1
12
2
1
1
4
2
2
4
1
12
4
1
1
4
1
12
2
10
1
1
As we can
4
As we can
4
1
2
1
1
2
1
4
1
1
2
As we can
2
2
2
1
1
2
4
1
4
1
2
4
4
1
2
2
1
1
1
2
1
4
2
2
1
4
1
As we can
4
2
2
4
1
15
4
1
2
2
1
As we can
1
1
15
1
2
2
2
2
As we can
3
As we can
2
1
4
16
12
4
1
2
1
1
12
12
2
As we can
1
4
2
3
4
4
2
10
1
2
21
2
4
1
1
2
4
2
4
10
4
As we can
1
4
3
As we can
4
1
1
4
1
2
28
2
1
3
2
1
2
3
10
3
1
2
2
2
1
1
16
2
1
2
4
3
2
3
1
3
2
2
4
1
2
1
2
4
2
4
8
4
As we can
1
4
2
12
1
4
2
2
As we can
4
1
2
1
2
As we can
1
As we can
2
1
2
3
1
4
1
3
12
1
4
4
2
1
1
1
1
1
4
2
3
As we can
As we can
4
1
1
As we can
12
As we can
2
1
16
4
4
3
2
15
4
2
2
2
1
4
2
2
1
4
3
12
12
4
2
1
2
18
1
12
1
4
3
2
2
1
2
1
2
1
2
2
4
1
2
1
1
2
3
1
10
1
1
1
2
4
2
1
12
4
18
3
12
30
12
4
4
2
2
1
As we can
2
3
1
4
1
3
2
1
4
1
2
2
1
2
2
2
As we can
1
2
21
2
1
2
2
2
3
1
As we can
10
1
2
2
1
2
4
4
4
4
As we can
2
4
2
1
2
2
As we can
1
4
12
1
2
2
As we can
2
2
4
12
2
As we can
2
2
12
4
1
2
2
3
2
2
1
1
4
2
2
2
1
As we can
2
3
2
1
1
2
2
12
1
2
1
2
